DatadogRUM/Sources/Instrumentation/WatchdogTerminations/WatchdogTerminationMonitor.swift

```diff
@@ -9,73 +9,137 @@ import DatadogInternal
 
 /// Monitors the Watchdog Termination events and reports them to Datadog.
 internal final class WatchdogTerminationMonitor {
+    /// The state of the Watchdog Termination Monitor.
+    enum State {
+        /// The monitor has started and is listening to the changes.
+        case started
+        /// The monitor is starting. It is still not listening to the changes.
+        case starting
+        /// The monitor has stopped and is not listening to the changes.
+        case stopped
+    }
+
     enum ErrorMessages {
         static let failedToCheckWatchdogTermination = "Failed to check if Watchdog Termination occurred"
         static let failedToStartAppState = "Failed to start Watchdog Termination App State Manager"
-        static let failedToStopAppState = "Failed to stop Watchdog Termination App State Manager"
         static let detectedWatchdogTermination = "Based on heuristics, previous app session was terminated by Watchdog"
-        static let detectedNonWatchdogTermination = "Previous app session was not terminated by Watchdog"
+        static let failedToReadViewEvent = "Failed to read the view event from the data store"
+        static let rumViewEventUpdated = "RUM View event updated"
+        static let failedToSendWatchdogTermination = "Failed to send Watchdog Termination event"
+        static let failedToDecodeLaunchReport = "Fails to decode LaunchReport in RUM"
     }
 
     let checker: WatchdogTerminationChecker
     let appStateManager: WatchdogTerminationAppStateManager
     let feature: FeatureScope
     let reporter: WatchdogTerminationReporting
+    weak var core: DatadogCoreProtocol?
```

Comment:
suggestion/ This looks against our goals from #1744 where we started removing direct core references from Features code. Direct refs are error prone (can leak if not weakified) and less scalable (they do not isolate core capabilities, instead bloat `DatadogCoreProtocol` with new things).

While I totally get the idea of having this reference here and `.mostRecentModifiedFileAt(before:)` doesn't fit into [`FeatureScope`](https://github.com/DataDog/dd-sdk-ios/blob/380bdf4938519b676d5a1d06bd2ce6928637a0a6/DatadogCore/Sources/Core/DatadogCore.swift#L311-L374), I wonder if we could avoid arbitrary `weak var core` by introducing `CoreUtils` type that manages the `core` reference similar to how it is done for [`CoreTelemetry`](https://github.com/DataDog/dd-sdk-ios/blob/380bdf4938519b676d5a1d06bd2ce6928637a0a6/DatadogInternal/Sources/Telemetry/Telemetry.swift#L366-L389). Just like with `Telemetry`, it can be safe to pass `CoreUtils` reference around and retain it on non-optional value.

cc @maxep what would be your take on this?

Reply:
This is recurrent problem and I think we can do better here, this perhaps beyond the scope of this PR but one idea that I have is to introduce something like
```swift
public struct Weak<T> {
    public var value: T?
    
    public init(value: T? = nil) {
        self.value = value
    }
}
```

and update the protocols to not conform AnyObject
```
public protocol DatadogCoreProtocol: MessageSending, BaggageSharing, CoreStorage {
```

this will be enforced at compiler level and we can use core using `Weak<DatadogCoreProtocol>`. This is must better approach than keep wrapping the core into different structs like we did for telemetry.

Let's sync on this.

Reply:
Action: for now, we do with wrapper mechanism as we do today and let's try to simplify after.